src/utils/data-access.js

```diff
@@ -35,3 +35,56 @@ export async function retrieveSiteBySiteId(dataAccess, siteId, log) {
     throw new Error(`Error getting site ${siteId}: ${e.message}`);
   }
 }
+
+/**
+ * Synchronizes existing suggestions with new data by removing outdated suggestions
+ * and adding new ones.
+ *
+ * @param {Object} params - The parameters for the sync operation.
+ * @param {Object} params.opportunity - The opportunity object to synchronize suggestions for.
+ * @param {Array} params.newData - Array of new data objects to sync.
+ * @param {Function} params.buildKey - Function to generate a unique key for each item.
+ * @param {Function} params.mapNewSuggestion - Function to map new data to suggestion objects.
+ * @param {Object} params.log - Logger object for error reporting.
+ * @returns {Promise<void>} - Resolves when the synchronization is complete.
+ */
+export async function syncSuggestions({
+  opportunity,
+  newData,
+  buildKey,
+  mapNewSuggestion,
+  log,
+}) {
+  const newDataKeys = new Set(newData.map(buildKey));
+  const existingSuggestions = await opportunity.getSuggestions();
+
+  // Remove outdated suggestions
```

Comment:
If the existing suggestion was entered via back-office-ui as the source, we should NOT delete/modify it, irrespective of whether the same suggestion exists in the new audit or not

Reply:
We will handles this as a next step. This will require an additional property in the data model